conductor-client/src/main/kotlin/com/openlattice/assembler/PostgresRoles.kt

```diff
@@ -13,14 +17,41 @@ private const val INTERNAL_PREFIX = "ol-internal"
  */
 class PostgresRoles private constructor() {
     companion object {
+
+        @JvmStatic
+        fun getOrCreatePermissionRole(
+                externalRoleNames: IMap<AccessTarget, UUID>,
+                tableId: UUID,
+                columnId: UUID,
+                permission: Permission
+        ): String {
+            return getOrCreatePermissionRole(externalRoleNames, AccessTarget.forPermissionOnTarget(permission, tableId, columnId))
+        }
+
+        @JvmStatic
+        fun getOrCreatePermissionRole( externalRoleNames: IMap<AccessTarget, UUID>, column: TableColumn, permission: Permission): String {
+            return getOrCreatePermissionRole(externalRoleNames, AccessTarget.forPermissionOnTarget(permission, column.tableId, column.columnId))
+        }
+
+        @JvmStatic
+        fun getOrCreatePermissionRole( externalRoleNames: IMap<AccessTarget, UUID>, permission: Permission, root: UUID, vararg parts: UUID ): String {
+            return getOrCreatePermissionRole(externalRoleNames, AccessTarget.forPermissionOnTarget(permission, root, *parts))
+        }
+
+        @JvmStatic
+        fun getOrCreatePermissionRole( externalRoleNames: IMap<AccessTarget, UUID>, target: AccessTarget ): String {
+            return externalRoleNames.executeOnKey(target, GetOrCreateExternalPermissionRoleNameEntryProcessor())
+        }
+
         @JvmStatic
         fun buildOrganizationUserId(organizationId: UUID): String {
             return "$INTERNAL_PREFIX|organization|$organizationId"
         }
 
         @JvmStatic
-        fun buildPostgresRoleName(role: Role): String {
-            return "$INTERNAL_PREFIX|role|${role.id}"
+        @Deprecated("Role names are stored in dbcredentials mapstore")
```

Comment:
I don't think these necessarily need to be deprecated, although they should only be used for initialization (probably as private functions wherever role initialization happens)

Reply:
(nvm, this isn't the one used for initialization)